internal/policy/policy.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2022 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package policy
+
+import (
+	"context"
+	"crypto"
+	"errors"
+	"strings"
+
+	ecc "github.com/hacbs-contract/enterprise-contract-controller/api/v1alpha1"
+	"github.com/sigstore/cosign/cmd/cosign/cli/rekor"
+	"github.com/sigstore/cosign/pkg/cosign"
+	cosignSig "github.com/sigstore/cosign/pkg/signature"
+	sigstoreSig "github.com/sigstore/sigstore/pkg/signature"
+	log "github.com/sirupsen/logrus"
+)
+
+// WithRekorUrl returns a new instance of EnterpriseContractPolicy if an update
+// to its rekorUrl is needed. Otherwise, the same instance is returned.
+func WithRekorUrl(policy *ecc.EnterpriseContractPolicy, rekorUrl string) *ecc.EnterpriseContractPolicy {
```

Comment:
What if the rekorUrl is passed on the cli and it's in the policy? Is the user notified the cli param is ignored? Or should the policy be overridden?

Reply:
I had a [discussion|https://issues.redhat.com/browse/HACBS-767?focusedCommentId=20778298&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-20778298] with myself about the precedence of these values. I reasoned that parameters from the CLI should overwrite whatever is in the ECP.

In `cmd/validate_image.go`, I changed the description of the rekorUrl and publicKey parameters to indicate this behavior. Does that address your concern?

Reply:
perfect!